gremlin-go/driver/translator_test.go

```diff
@@ -0,0 +1,669 @@
+			},
+			equals: "g.V().hasLabel('airport').where(out().count().is(gt(250))).values('code')",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.V().HasLabel("airport").Filter(T__.Out().Count().Is(P.Gt(250))).Values("code")
+			},
+			equals: "g.V().hasLabel('airport').filter(out().count().is(gt(250))).values('code')",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.WithSack(0).
+					V("3").
+					Repeat(T__.OutE("route").Sack(Operator.Sum).By("dist").InV()).
+					Until(T__.Has("code", "AGR").Or().Loops().Is(4)).
+					Has("code", "AGR").
+					Local(T__.Union(T__.Path().By("code").By("dist"), T__.Sack()).Fold()).
+					Limit(10)
+			},
+			equals: "g.withSack(0).V('3').repeat(outE('route').sack(sum).by('dist').inV()).until(has('code','AGR').or().loops().is(4)).has('code','AGR').local(union(path().by('code').by('dist'),sack()).fold()).limit(10)",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.AddV().As("a").AddV().As("b").AddE("knows").From("a").To("b")
+			},
+			equals: "g.addV().as('a').addV().as('b').addE('knows').from('a').to('b')",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.AddV("Person").As("a").AddV("Person").As("b").AddE("knows").From("a").To("b")
+			},
+			equals: "g.addV('Person').as('a').addV('Person').as('b').addE('knows').from('a').to('b')",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.V("3").Project("Out", "In").By(T__.Out().Count()).By(T__.In().Count())
+			},
+			equals: "g.V('3').project('Out','In').by(out().count()).by(in().count())",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.V("44").Out().Aggregate("a").Out().Where(P.Within("a")).Path()
+			},
+			equals: "g.V('44').out().aggregate('a').out().where(within('a')).path()",
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.V().Has("date", time.Date(2021, 2, 22, 0, 0, 0, 0, time.UTC))
+			},
+			equals: fmt.Sprintf("g.V().has('date',%s)", time.Date(2021, 2, 22, 0, 0, 0, 0, time.UTC).String()),
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.V().Has("date", P.Within(time.Date(2021, 2, 22, 0, 0, 0, 0, time.UTC), time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC)))
+			},
+			equals: fmt.Sprintf("g.V().has('date',within([%s,%s]))", time.Date(2021, 2, 22, 0, 0, 0, 0, time.UTC).String(), time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC).String()),
+		},
+		{
+			assert: func(g *GraphTraversalSource) *GraphTraversal {
+				return g.V().Has("date", P.Between(time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC), time.Date(2021, 2, 22, 0, 0, 0, 0, time.UTC)))
+			},
+			equals: fmt.Sprintf("g.V().has('date',between([%s,%s]))", time.Date(2021, 1, 1, 0, 0, 0, 0, time.UTC).String(), time.Date(2021, 2, 22, 0, 0, 0, 0, time.UTC).String()),
```

Comment:
This doesn't seem like the right translation for dates. This doesn't execute as expected in the console for me. I would expect a translation more like
```g.V().has('date',between(new Date(121,1,1,0,0,0),new Date(121,2,22,0,0,0)))```